use-shopping-cart/src/reducers.js

```diff
@@ -95,16 +136,33 @@ export function cartValuesReducer(state, action) {
     }
   }
 
-  function updateEntry(sku, count) {
+  function updateEntry(sku, count, price_metadata, product_metadata) {
     const cartDetails = { ...state.cartDetails }
     const entry = cartDetails[sku]
     if (entry.quantity + count <= 0) return removeEntry(sku)
 
+    if (!entry.price_data && price_metadata) {
+      entry.price_data = {}
+
+      entry.price_data = {
+        ...price_metadata
+      }
+    }
+
+    if (entry.price_data && price_metadata) {
+      entry.price_data = {
+        ...entry.price_data,
+        ...price_metadata
+      }
+    }
```

Comment:
Same thing as above

Reply:
Yeah, I just want clarification on this redundancy. I had to account for products that started off with no `price_data` in the initial object and then price_data gets added later on with the new method. Would definitely like to see a better approach.

Reply:
Same issues as line 76 through 85. Other than that, this code isn't really too redundant.